src/test/java/com/KacFlor/ShopSpring/service/CustomerServiceTest.java

```diff
@@ -0,0 +1,188 @@
+package com.KacFlor.ShopSpring.service;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import com.KacFlor.ShopSpring.controllersRequests.CustomerUpdateRequest;
+import com.KacFlor.ShopSpring.dao.CustomerRepository;
+import com.KacFlor.ShopSpring.dao.UserRepository;
+import com.KacFlor.ShopSpring.model.Customer;
+import com.KacFlor.ShopSpring.model.Role;
+import com.KacFlor.ShopSpring.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.*;
+
+import org.springframework.security.core.Authentication;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+import java.util.List;
+import java.util.Optional;
+
+@ExtendWith(MockitoExtension.class)
+public class CustomerServiceTest{
+
+    @Mock
+    private CustomerRepository customerRepository;
+
+    @Mock
+    private UserRepository userRepository;
+
+    @InjectMocks
+    private CustomerService customerService;
+
+    private User user1;
```

Comment:
Why are these variables needed, where are they used?

Reply:
i create user1,user2 and hAdmin to simulate register process in project

Reply:
look my comment above

Reply:
done